apps/meteor/app/livechat/server/api/lib/livechat.ts

```diff
@@ -25,18 +25,23 @@ async function findTriggers(): Promise<Pick<ILivechatTrigger, '_id' | 'actions'
 
 async function findDepartments(
 	businessUnit?: string,
+	userId?: string,
 ): Promise<Pick<ILivechatDepartment, '_id' | 'name' | 'showOnRegistration' | 'showOnOfflineForm' | 'departmentsAllowedToForward'>[]> {
 	// TODO: check this function usage
 	return (
 		await LivechatDepartment.findEnabledWithAgentsAndBusinessUnit<
 			Pick<ILivechatDepartment, '_id' | 'name' | 'showOnRegistration' | 'showOnOfflineForm' | 'departmentsAllowedToForward'>
-		>(businessUnit, {
-			_id: 1,
-			name: 1,
-			showOnRegistration: 1,
-			showOnOfflineForm: 1,
-			departmentsAllowedToForward: 1,
-		})
+		>(
+			businessUnit,
+			{
+				_id: 1,
+				name: 1,
+				showOnRegistration: 1,
+				showOnOfflineForm: 1,
+				departmentsAllowedToForward: 1,
+			},
+			{ userId },
```

Comment:
What if, instead of passing the userId to the model, we validate the units before calling the `findEnabled...` function?